surfacers/postgres/postgres.go

```diff
@@ -0,0 +1,295 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package postgres implements "postgres" surfacer. This surfacer type is in
+// experimental phase right now.
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+
+	"database/sql"
+	"encoding/json"
+	configpb "github.com/google/cloudprober/surfacers/postgres/proto"
+	"github.com/lib/pq"
+	"strconv"
+	"time"
+)
+
+type label struct {
+	key   string
+	value string
+}
+
+type pgMetric struct {
+	time       time.Time
+	metricName string
+	value      string
+	labels     []label
+}
+
+// labelsJSON takes the label array and formats it for insertion into
+// postgres jsonb labels column, storing each label as k,v json object
+func (c pgMetric) labelsJSON() (string, error) {
+	m := make(map[string]string)
+	for _, l := range c.labels {
+		m[l.key] = l.value
+	}
+
+	bs, err := json.Marshal(m)
+	if err != nil {
+		return "", err
+	}
+
+	return string(bs), nil
+}
+
+func newPGMetric(t time.Time, metricName, val string, labels []label) pgMetric {
+	return pgMetric{
+		time:       t,
+		metricName: metricName,
+		value:      val,
+		labels:     labels,
+	}
+}
+
+type pgDistribution struct {
+	*metrics.DistributionData
+
+	metricName string
+	labels     []label
+	timestamp  time.Time
+}
+
+// sumMetric creates the correctly named metric
+// representing sum of the distribution
+func (d pgDistribution) sumMetric() pgMetric {
+	return newPGMetric(
+		d.timestamp,
+		d.metricName+"_sum",
+		strconv.FormatFloat(d.Sum, 'f', -1, 64),
+		d.labels,
+	)
+}
+
+// countMetric creates the correctly named metric representing
+// the count of the distribution
+func (d pgDistribution) countMetric() pgMetric {
+	return newPGMetric(
+		d.timestamp,
+		d.metricName+"_count",
+		strconv.FormatInt(d.Count, 10),
+		d.labels,
+	)
+}
+
+// bucketMetrics creates and formats all metrics for each bucket in this distribution.
+// each bucket is assigned a metric name suffixed with "_bucket" and labeled with the
+// corresponding bucket as "le: {bucket}"
+func (d pgDistribution) bucketMetrics() []pgMetric {
+	var val int64
+	ms := []pgMetric{}
+
+	for i := range d.LowerBounds {
+		val += d.BucketCounts[i]
+		var lb string
+		if i == len(d.LowerBounds)-1 {
+			lb = "+Inf"
+		} else {
+			lb = strconv.FormatFloat(d.LowerBounds[i+1], 'f', -1, 64)
+		}
+		labelsWithBucket := append(d.labels, label{"le", lb})
+		ms = append(ms, newPGMetric(d.timestamp, d.metricName+"_bucket", strconv.FormatInt(val, 10), labelsWithBucket))
+	}
+
+	return ms
+}
+
+// metricRows extracts all metrics to be insterted into postgres
+// corresponding to the EventMEtric
+func metricRows(em *metrics.EventMetrics) []pgMetric {
+	fmt.Printf("%+v\n", em)
+	cs := []pgMetric{}
```

Comment:
I couldn't quite follow what "cs" could stand for? Can you either clarify or use something more generic -- rows, metrics, etc.